HAL.md

```diff
@@ -0,0 +1,110 @@
+# Hardware abstract layer(HAL) overview
+
+## Abstract
+Hardware abstract layer(HAL) defines a standard interface for hardware vendors to implement, which enables Tangle-Accelerator to be aware of lower-level driver implementations. HAL layer supports such as uart, GPIO, secure storage and other low-level operators inside endpoint. The endpoint is a daemon, offering monitor for the device and can send the message to tangle-accelerator. HAL layer lets the user only need to implement their device operators from abstract interface which defined inside `device.h` but not need to know how endpoint work.
+
+## Directory overview
+* connectivity: Connect to Tangle-accelerator functions.
+* devices: The real directory to implement devices.
+* hal: Hardware abstraction layer.
+* hooks: Git hook functions.
+* tests: Unit-tests.
+* third_party: Third party directory.
+* utils: Utility functions for TA-endpoint.
+
+## How to implement new device
+
+Create directory for the new device under `devices`.
+```
+mkdir devices/mydevice
+```
+* Create `impl.c` and `Makefile` under new device directory.
+* Include `device.h` headers into new device header file.
```

Comment:
Since you include `device.h` in the `impl.c` (as shown in the example), the description need to be modified from header file to source file.